src/main/java/org/mockito/internal/configuration/SpyAnnotationEngine.java

```diff
@@ -89,7 +90,13 @@ private static Object newSpyInstance(Object testInstance, Field field)
         if (type.isInterface()) {
             return Mockito.mock(type, settings.useConstructor());
         }
-        if (!Modifier.isStatic(type.getModifiers())) {
+        int modifiers = type.getModifiers();
+        if (Modifier.isPrivate(modifiers) && Modifier.isAbstract(modifiers) && type.getEnclosingClass() != null) {
+            throw new MockitoException("@Spy annotation can't initialize private abstract inner classes. "
```

Comment:
Is it possible to offer a suggestion to the user? For example, make the class public, etc.